tools/mininet/tofino-model.py

```diff
@@ -0,0 +1,253 @@
+
+            # We want to be notified if processes quits prematurely...
+            self.stopped = False
+            threading.Thread(target=watchdog, args=[self]).start()
+
+        except Exception:
+            StratumTofinoModel.mininet_exception = 1
+            self.stop()
+            raise
+
+    def getPortVethMap(self):
+        intf_number = 1
+        portsVeth = list()
+        for intf_name in self.intfNames():
+            if intf_name == 'lo': continue
+            portsVeth.append(
+                {
+                    "device_port": intf_number,
+                    "veth1": int(intf_name[4:]),
+                    "veth2": int(intf_name[4:])+100
+                }
+            )
+            intf_number+=1
+
+        data =  { "PortToVeth": portsVeth }
+        return json.dumps(data, indent=4)+"\n"
+
+    def getChassisConfig(self):
+        config = """description: "chassis config bf tofino model {name}"
+chassis {{
+  platform: PLT_GENERIC_BAREFOOT_TOFINO
+  name: "{name}"
+}}
+nodes {{
+  id: {nodeId}
+  name: "{name} node {nodeId}"
+  slot: 1
+  index: 1
+}}\n""".format(name=self.name, nodeId=self.nodeId)
+
+        intf_number = 1
+        for intf_name in self.intfNames():
+            if intf_name == 'lo':
+                continue
+            config = config + """singleton_ports {{
+  id: {intfNumber}
+  name: "{intfName}"
+  slot: 1
+  port: {intfNumber}
+  channel: 1
+  speed_bps: 10000000000
+  config_params {{
+    admin_state: ADMIN_STATE_ENABLED
+  }}
+  node: {nodeId}
+}}\n""".format(intfName=intf_name, intfNumber=intf_number, nodeId=self.nodeId)
+            intf_number += 1
+
+        return config
+
+    def procHandleLive(self):
+        if self.t_modelProc_h is not None: return True
+        if self.stratumProc_h is not None: return True
+        return False
+
+    def tryTerminate(self):
+        if self.t_modelProc_h is not None:
+            if self.t_modelProc_h.poll() is None:
+                self.t_modelProc_h.terminate()
+                self.t_modelProc_h.wait()
+        if self.stratumProc_h is not None:
+            if self.stratumProc_h.poll() is None:
+                self.stratumProc_h.terminate()
+                self.stratumProc_h.wait()
+        if self.t_logfd is not None:
+            self.t_logfd.close()
+            self.t_logfd = None
+        if self.s_logfd is not None:
+            self.s_logfd.close()
+            self.s_logfd = None
+
+class NoOffloadHost(Host):
+    def __init__(self, name, inNamespace=True, **params):
+        Host.__init__(self, name, inNamespace=inNamespace, **params)
+
+    def config(self, **params):
+        r = super(Host, self).config(**params)
+        for off in ["rx", "tx", "sg"]:
+            cmd = "/sbin/ethtool --offload %s %s off" \
+                  % (self.defaultIntf(), off)
+            self.cmd(cmd)
+        return r
+
+class TofinoLink( Link ):
```

Comment:
```suggestion
class TofinoLink(Link):
```
also apply to others below